File: Backend/models/ProgramPlan.js
const Sequelize = require('sequelize');
const sequelize = require('../config/database');
const Course = require('./Course');

const ProgramPlan = sequelize.define('Program_Plans', {
  plan_id: {
    type: Sequelize.INTEGER,
    autoIncrement: true,
    primaryKey: true,
  },
  program_id: {
    type: Sequelize.STRING(10),
    allowNull: false,
  },
  course_id: {
    type: Sequelize.STRING(10),
    allowNull: true,
  },
  category: {
    type: Sequelize.STRING(50),
    allowNull: true,
  },
}, {
  timestamps: false,
  tableName: 'Program_Plans',
});

module.exports = ProgramPlan;

File: grade-simulator-service/dummy_data/models/program_plan.py
from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship
from db import Base

class ProgramPlan(Base):
    __tablename__ = "Program_Plans"
    
    plan_id = Column(Integer, primary_key=True, autoincrement=True)
    program_id = Column(String(10), nullable=False)
    course_id = Column(String(10), ForeignKey("Courses.course_id"), nullable=True)
    category = Column(String(50), nullable=True)
    
    def to_dict(self):
        return {
            "plan_id": self.plan_id,
            "program_id": self.program_id,
            "course_id": self.course_id,
            "category": self.category
        } 